lib/good_job/scheduler.rb

```diff
@@ -137,11 +137,7 @@ def create_thread(state = nil)
         return false unless @performer.next?(state)
       end
 
-      future = Concurrent::Future.new(args: [@performer], executor: @pool) do |performer|
-        output = nil
-        Rails.application.executor.wrap { output = performer.next }
-        output
-      end
+      future = Concurrent::Future.new(executor: @pool, &@performer.method(:next))
```

Comment:
This is very clean, but I think I may regret it in the future because it's easy to overlook _this is where the work gets done_.

But I'm happy to roll with it for now and just make a mental note that if it causes confusion in the future, to go to the more expansive version.

And just while I'm noodling:
- I think it would be possible to cache the proc so it doesn't have to be recreated everytime this method is called
- I wonder if the performer should alias `call` to `next` so it is directly callable (I'm imagining there was a reason why that wasn't done originally but I'm not remembering it... though it was likely that I wanted explicitness)